### ISSUE TITLE
What is the difference between ControlNet and concatenating conditional frames with noisy latents?

### ISSUE THREAD

**Issue by YuyangYin** [TheDenk/cogvideox-controlnet#26]
Hi, I've noticed that in the original CogVideoI2V model, it concatenates the latent and image_latents in the channel dimension. Here is the relevant code:
```
 padding_shape = (latent.shape[0], latent.shape[1] - 1, *latent.shape[2:])
 latent_padding = image_latents.new_zeros(padding_shape)
image_latents = torch.cat([image_latents, latent_padding], dim=1)
latent_noisy = self.components.scheduler.add_noise(latent, noise, timesteps)
latent_img_noisy = torch.cat([latent_noisy, image_latents], dim=2)
```

So I'm wondering, if I change the padding zeros to conditional frames, what would be the difference compared to the version with ControlNet? Does anyone know? Thanks.

**Comment by TheDenk** [comment]
Hey :)
The original model was trained with padding and if you change the padding zeros to conditional frame most likely it will work quite worse.

**Comment by OrangeSodahub** [comment]
@TheDenk Do you have any thoughs about why it would be worse?

**Comment by TheDenk** [comment]
> The original model was trained with padding

@OrangeSodahub because of this
